composer/loggers/tensorboard_logger.py

```diff
@@ -0,0 +1,102 @@
+            every 10 batches. The logs will also be automatically flushed at the start and
+            end of every evaluation phase (`EVENT.EVAL_START` and `EVENT.EVAL_END` ),
+            the end of every epoch (`EVENT.EPOCH_END`), and the end of training
+            (`EVENT.FIT_END`). Default: ``100``.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            Recommended to be true since the rank 0 will have access to most global metrics.
+            A setting of `False` may lead to logging of duplicate values.
+            Default: :attr:`True`.
+    """
+
+    def __init__(self,
+                 run_name: Optional[str] = None,
+                 log_dir: Optional[str] = None,
+                 flush_interval: int = 100,
+                 rank_zero_only: bool = True):
+
+        self.run_name = run_name
+        self.log_dir = log_dir
+        self.flush_interval = flush_interval
+        self.rank_zero_only = rank_zero_only
+        self.artifact_name: str = None
+        self.writer: Optional[SummaryWriter] = None
+
+    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
+        del log_level
+
+        if self.rank_zero_only and dist.get_global_rank() != 0:
+            return
+        for tag, data_point in data.items():
+            if isinstance(data_point, str):  # Will error out with weird caffe2 import error.
+                continue
+            try:
+                self.writer.add_scalar(tag, data_point, global_step=int(state.timestamp.batch))
+            except NotImplementedError:
+                pass
+
+    def init(self, state: State, logger: Logger) -> None:
+        if self.run_name is None:
+            self.run_name = f'{state.run_name}'
+        if self.log_dir is None:
+            self.log_dir = 'tensorboard_logs'
+        summary_writer_log_dir = Path(self.log_dir) / self.run_name
+        self.writer = SummaryWriter(log_dir=summary_writer_log_dir,
+                                    filename_suffix=self.run_name)
+
+    def batch_end(self, state: State, logger: Logger) -> None:
+        if int(state.timestamp.batch) % self.flush_interval == 0:
+            self._flush(logger)
+
+    def epoch_end(self, state: State, logger: Logger) -> None:
+        self._flush(logger)
+
+    def eval_start(self, state: State, logger: Logger) -> None:
+        self._flush(logger)
+
+    def eval_end(self, state: State, logger: Logger) -> None:
+        self._flush(logger)
+
+    def fit_end(self, state: State, logger: Logger) -> None:
+        # Flush the file on fit_end, in case if was not flushed on epoch_end and the trainer is re-used
+        # (which would defer when `self.close()` would be invoked)
+        self._flush(logger)
+
+    def _flush(self, logger: Logger):
+        self.writer.flush()
+        logger.file_artifact(LogLevel.FIT,
+                             artifact_name=self.run_name,
```

Comment:
Can there be an artifact name prefix config variable, that defaults to '{run_name}/tensorboard_logs'?
The artifact name here should include the global rank of the process, so each rank will write to its own file.

Reply:
it isn't necessary to add the global rank since the `file_artifact` function does that for you here: https://github.com/mosaicml/composer/blob/dev/composer/loggers/logger.py#L123

Reply:
that is a good point that I should probably name the `artifact_name=Path(self.writer.file_writer.event_writer._file_name).stem` or something because it is an actual tensorboard log file that gets saved as an artifact

Reply:
Actually, Tensorboard won't read the log file unless it is named properly so calling it `Path(self.writer.file_writer.event_writer._file_name).stem` is essential. Fixed here: 
42426e6

Reply:
update: I use the formatted string like so: `'tensorboard_logs/{run_name}/events.out.tfevents-{run_name}-{rank}'`